drivers/sensor/lis2dw12/lis2dw12_trigger.c

```diff
@@ -192,11 +192,10 @@ static void lis2dw12_gpio_callback(struct device *dev,
 {
 	struct lis2dw12_data *lis2dw12 =
 		CONTAINER_OF(cb, struct lis2dw12_data, gpio_cb);
-	const struct lis2dw12_device_config *cfg = dev->config->config_info;
 
 	ARG_UNUSED(pins);
 
-	gpio_pin_disable_callback(dev, cfg->int_gpio_pin);
+	gpio_pin_disable_callback(dev, DT_INST_0_ST_LIS2DW12_IRQ_GPIOS_PIN);
```

Comment:
Hardcoding the pin number here makes it harder to have multiple instances of the driver. Instead you could store the pin number in `lis2dw12_data` during init and grab it from there in the callback. The fxos8700 driver does this.
https://github.com/zephyrproject-rtos/zephyr/blob/1593de9642fc0e6509b22862b570c5c4d0618818/drivers/sensor/fxos8700/fxos8700_trigger.c#L14-L25

Reply:
You are right. I have also added similar pin checking condition as in fxos8700. Changes have been pushed.

Reply:
Yes, actually this was my point yesterday and I felt there was a reason behind it. Thank you @mhelm for your comment.